source/extensions/filters/http/istio_stats/istio_stats.cc

```diff
@@ -1206,10 +1219,10 @@ class IstioStatsFilter : public Http::PassThroughFilter,
         tags_.push_back({context_.destination_service_name_, service_host_name.empty()
                                                                  ? context_.canonical_name_
                                                                  : pool_.add(service_host_name)});
+        tags_.push_back({context_.destination_cluster_, context_.cluster_name_});
+        tags_.push_back({context_.destination_service_namespace_, context_.namespace_});
```

Comment:
This looks like it's for the client sidecar path not servergateway (waypoint)

Reply:
Before the code was shared for the two. The waypoint part was wrong, so I moved the shared code (prev L1211) into the sidecar path and changed the waypoint code.

Also its ServerSidecar btw, its a double nested `switch` which is slightly confusing

Reply:
Ah, I see - I misread it in the GitHub diff. Looks like a lot of tests failed though

Reply:
accidentally re-ordered the lines in the move (which I think is a change that is a NOP except in tests, fwiw). Should be good now.